File: Ian-Neves/atividade-1.cpp
/*Construa uma função para calcular a soma de dois números inteiros.

Exemplo
a = 7
b = 3

Retorne 10.

Descrição da função
Complete a função com os seguintes parâmetros:

int a: o primeiro valor
int b: o segundo valor

Retorno
int: a soma de a e b

Restrições
1 ≤ a, b ≤ 1000

Entrada de exemplo
a = 2
b = 3
Saída de exemplo
5
Explicação
2 + 3 = 5.
*/

#include <iostream>
using namespace std;

int soma(int a, int b) {
    return a + b;
}

int main() {
    int a, b;

    cin >> a;
    cin >> b;

    cout << soma(a, b) << endl;

    return 0;
}

File: Bruno-Correia/atividade-1.py
def main():
    ''''
    # Atividade 1 — Soma de dois números inteiros
    ## Descrição
    - Função que recebe dois números inteiros e retorna a soma entre eles.

    ## Parâmetros
    | Parâmetro | Tipo | Descrição       |
    |-----------|------|-----------------|
    | `a`       | int  | O primeiro valor |
    | `b`       | int  | O segundo valor  |

    ## Retorno
    Do tipo `int` retorna a soma de a e b.

    ## Restrições
    - `1 ≤ a, b ≤ 1000`
    - Caso algum valor esteja fora do intervalo, retorna um erro.
    '''
    a = int(input("a = "))
    b = int(input("b = "))

    if not 1 <= a <= 1000:
        raise ValueError("Os valores devem estar entre 1 e 1000.")
    
    if not 1 <= b <= 1000:
        raise ValueError("Os valores devem estar entre 1 e 1000.")

    result = a + b
    print(result)

if __name__ == "__main__":
    main()